### PR TITLE
feat(web-components): addition of web components to website

### PR DESCRIPTION
**Pull request by jeffchew**
This PR introduced documentation around the support of Carbon Web
Components. This is a community contributed library that is based on the
 Web Components custom elements and shadow dom spec, and is the new
 browser standard.

#### Changelog

**New**

- Web Components page under `developing`

**Changed**

- Added references to web components storybook/github

### CODE REVIEW

**Comment by jeffchew** [comment]
I have read the DCO document and I hereby sign the DCO.

**Review by jnm2377 — APPROVED** [review]

File: src/pages/developing/frameworks/web-components.mdx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    'Vue',
+    'Svelte',
+    'Web Components',
+    'Other frameworks',
+  ]
+---
+
+<PageDescription>
+
+The library provides front-end developers & engineers a collection of reusable
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:20]:
```suggestion
The library provides front-end developers and engineers a collection of reusable
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+### Basic usage
+
+Our example at [CodeSandbox](https://codesandbox.io) shows the most basic usage:
+
+[![Edit carbon-web-components](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/carbon-design-system/carbon-web-components/tree/master/examples/codesandbox/basic)
+
+The first thing you need is **setting up a module bundler** to resolve
+ECMAScript `import`s. Above example uses [Webpack](https://webpack.js.org). You
+can use other bundlers like [Rollup](https://rollupjs.org/), too.
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:84]:
```suggestion
can use other bundlers like [Rollup](https://rollupjs.org/) too.
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+Once you set up a module bundler, you can start importing our component modules,
+like:
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:87]:
```suggestion
for example:
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+```javascript
+import 'carbon-web-components/es/components/dropdown/dropdown.js';
+import 'carbon-web-components/es/components/dropdown/dropdown-item.js';
+```
+
+Once you do that, you can use our components in the same manner as native HTML
+tags, like:
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:95]:
```suggestion
tags, for example:
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+<bx-dropdown trigger-content="Select an item">
+  <bx-dropdown-item value="all">Option 1</bx-dropdown-item>
+  <bx-dropdown-item value="cloudFoundry">Option 2</bx-dropdown-item>
+  <bx-dropdown-item value="staging">Option 3</bx-dropdown-item>
+  <bx-dropdown-item value="dea">Option 4</bx-dropdown-item>
+  <bx-dropdown-item value="router">Option 5</bx-dropdown-item>
+</bx-dropdown>
+```
+
+If you just want to try our components for demonstrations, etc., you can use
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:107]:
```suggestion
If you just want to try our components for demonstrations and so on, you can use
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+CDNs that support module mapping (e.g. [JSPM](https://jspm.org)). With it, you
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:108]:
```suggestion
CDNs that support module mapping (for example, [JSPM](https://jspm.org)). With it, you
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+can just import our modules in `<script type="module">`:
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:109]:
```suggestion
can import just our modules in `<script type="module">`:
```

**Review by janchild — APPROVED** [review]
@jeffchew just a few small copyedits but this looks great!

**Comment by jeffchew** [comment]
> @jeffchew just a few small copyedits but this looks great!

Thank you @janchild for the suggestions!

@asudoh we should probably apply the same copy edits to the main README in the `carbon-web-components` repo too.

**Review by mattrosno — CHANGES_REQUESTED** [review]
We need to somewhere state the difference between the React components and the React wrapper of the web components.

I'm not seeing the React wrapper mentioned in this PR, but should somebody go to the Storybook, see the React link in the Welcome story, and head to the "carbon-web-components with React" Storybook, there could be a lot of confusion as to why those React components are different then the canonical React components.

https://carbon-custom-elements-react.netlify.app/?path=/story/introduction-welcome--page

**Comment by jeffchew** [comment]
> We need to somewhere state the difference between the React components and the React wrapper of the web components.
> 
> I'm not seeing the React wrapper mentioned in this PR, but should somebody go to the Storybook, see the React link in the Welcome story, and head to the "carbon-web-components with React" Storybook, there could be a lot of confusion as to why those React components are different then the canonical React components.
> 
> https://carbon-custom-elements-react.netlify.app/?path=/story/introduction-welcome--page



> We need to somewhere state the difference between the React components and the React wrapper of the web components.
> 
> I'm not seeing the React wrapper mentioned in this PR, but should somebody go to the Storybook, see the React link in the Welcome story, and head to the "carbon-web-components with React" Storybook, there could be a lot of confusion as to why those React components are different then the canonical React components.
> 
> https://carbon-custom-elements-react.netlify.app/?path=/story/introduction-welcome--page

Good call @mattrosno ! I added a section regarding the framework wrappers and was explicit for its usage.

@asudoh , this again may be something we should update in the `carbon-web-components` README too.

**Review by emyarod — COMMENTED** [review]
on a side note, were the formatting changes in the other MDX files intentional? I believe `yarn format` may override them and reduce the size of the PR diff

**Review by emyarod — APPROVED** [review]
regarding the formatting previous formatting note, I may have different settings locally but otherwise looks good to me

**Comment by janchild** [comment]
Hi @jeffchew, I'm looking through this content again and I see that "Web Components" is capitalized and it should be written as "web components". Apologies for not catching in the first review. The IBM Style site has rules around capitalization and it's reserved for proper nouns or trademarks only. I can take a pass through the files for you. Just wanted to give you the heads up.

**Comment by jeffchew** [comment]
> Hi @jeffchew, I'm looking through this content again and I see that "Web Components" is capitalized and it should be written as "web components". Apologies for not catching in the first review. The IBM Style site has rules around capitalization and it's reserved for proper nouns or trademarks only. I can take a pass through the files for you. Just wanted to give you the heads up.

Thanks @janchild , makes sense! Does that also apply for when we call the repo `Carbon Web Components`, where we have to have them lower case to `Carbon web components`?

I can do a find replace to see if I capture it all at once.

**Comment by jeffchew** [comment]
> > Hi @jeffchew, I'm looking through this content again and I see that "Web Components" is capitalized and it should be written as "web components". Apologies for not catching in the first review. The IBM Style site has rules around capitalization and it's reserved for proper nouns or trademarks only. I can take a pass through the files for you. Just wanted to give you the heads up.
> 
> Thanks @janchild , makes sense! Does that also apply for when we call the repo `Carbon Web Components`, where we have to have them lower case to `Carbon web components`?
> 
> I can do a find replace to see if I capture it all at once.

@janchild FYI, I looked for standalone instances of `web components` and made them lowercase, I left instances of `Carbon Web Components` for now, though I was looking for instances of `Carbon Vanilla` and see that it's capitalized in that context here:

https://github.com/carbon-design-system/carbon-website/blame/main/src/pages/index.mdx#L87

**Comment by janchild** [comment]
> > > Hi @jeffchew, I'm looking through this content again and I see that "Web Components" is capitalized and it should be written as "web components". Apologies for not catching in the first review. The IBM Style site has rules around capitalization and it's reserved for proper nouns or trademarks only. I can take a pass through the files for you. Just wanted to give you the heads up.
> > 
> > 
> > Thanks @janchild , makes sense! Does that also apply for when we call the repo `Carbon Web Components`, where we have to have them lower case to `Carbon web components`?
> > I can do a find replace to see if I capture it all at once.
> 
> @janchild FYI, I looked for standalone instances of `web components` and made them lowercase, I left instances of `Carbon Web Components` for now, though I was looking for instances of `Carbon Vanilla` and see that it's capitalized in that context here:
> 
> https://github.com/carbon-design-system/carbon-website/blame/main/src/pages/index.mdx#L87

That works for me! Thanks, @jeffchew!

File: src/pages/developing/frameworks/web-components.mdx [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+</html>
+```
+
+### List of available components
+
+View available web components
+[here](https://web-components.carbondesignsystem.com). Usage information is
+available in the notes provided with each story.
+
+### JavaScript framework integration
```
Comment (mattrosno) [inline src/pages/developing/frameworks/web-components.mdx:152]: Can we be a little more direct with how the web component wrappers are different than Carbon Angular, Carbon React, and Carbon Vue component libraries? It's more than just who maintains them - it's differences in APIs (e.g. we wouldn't want somebody trying to reference the Carbon React docs when actually using the web component wrapper.) It's differences in accessibility audits and tests and to some degree, differences in behavior as well.

@joshblack have an suggestion on a disclaimer that we can put in a notification box here or something so there is no confusion between framework-native implementations, and framework wrappers of the web components?

**Review by mattrosno — CHANGES_REQUESTED** [review]

File: src/pages/developing/frameworks/web-components.mdx (hunk 8 above)
Comment (jeffchew) [reply src/pages/developing/frameworks/web-components.mdx:152]: @mattrosno I added a note in the section that highlights the difference, let me know if the wording works!

fyi @janchild

Comment (mattrosno) [reply src/pages/developing/frameworks/web-components.mdx:152]: Thanks!

**Review by mattrosno — APPROVED** [review]

File: src/pages/developing/frameworks/web-components.mdx [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+yarn add carbon-web-components carbon-components lit-html lit-element
+```
+
+### Basic usage
+
+Our example at [CodeSandbox](https://codesandbox.io) shows the most basic usage:
+
+[![Edit carbon-web-components](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/carbon-design-system/carbon-web-components/tree/master/examples/codesandbox/basic)
+
+The first thing you need is **setting up a module bundler** to resolve
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:82]:
```suggestion
The first thing you need to do is **set up a module bundler** to resolve
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+ECMAScript `import`s. Above example uses [Webpack](https://webpack.js.org). You
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:83]:
```suggestion
ECMAScript `import`s. The above example uses [Webpack](https://webpack.js.org) but you
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+
+Once you set up a module bundler, you can start importing our component modules,
+for example:
+
+```javascript
+import 'carbon-web-components/es/components/dropdown/dropdown.js';
+import 'carbon-web-components/es/components/dropdown/dropdown-item.js';
+```
+
+Once you do that, you can use our components in the same manner as native HTML
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:94]:
```suggestion
Once you've imported the component modules, you can use our components in the same manner as native HTML
```

File: src/pages/developing/frameworks/web-components.mdx [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+  <bx-dropdown-item value="cloudFoundry">Option 2</bx-dropdown-item>
+  <bx-dropdown-item value="staging">Option 3</bx-dropdown-item>
+  <bx-dropdown-item value="dea">Option 4</bx-dropdown-item>
+  <bx-dropdown-item value="router">Option 5</bx-dropdown-item>
+</bx-dropdown>
+```
+
+If you just want to try our components for demonstrations and so on, you can use
+CDNs that support module mapping (for example, [JSPM](https://jspm.org)). With
+it, you can import just our modules in `<script type="module">`:
```
Comment (janchild) [inline src/pages/developing/frameworks/web-components.mdx:109]:
```suggestion
CDNs, you can import just our modules in `<script type="module">`:
```

**Review by janchild — COMMENTED** [review]
@jeffchew I read through again today, and had just a couple more copyediting tweaks. After this, it looks like it's good to merge.

**Comment by jeffchew** [comment]
> @jeffchew I read through again today, and had just a couple more copyediting tweaks. After this, it looks like it's good to merge.

Thank you for all of the suggested changes!

**Comment by janchild** [comment]
> Thank you for all of the suggested changes!

You're very welcome! Would you like me to merge, @jeffchew?

**Comment by jeffchew** [comment]
> > Thank you for all of the suggested changes!
> 
> You're very welcome! Would you like me to merge, @jeffchew?

Yes, would be fantastic! I wasn't sure what the rules or process was for merging 😄

**Pull request merged**